src/UTMSP/UTMSPMapPolygonVisuals.qml

```diff
@@ -0,0 +1,708 @@
+ *
+ * (c) 2023 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
+import QtQuick                          2.11
+import QtQuick.Controls                 2.4
+import QtLocation                       5.3
+import QtPositioning                    5.3
+import QtQuick.Dialogs                  1.2
+import QtQuick.Layouts                  1.11
+
+import QGroundControl                   1.0
+import QGroundControl.ScreenTools       1.0
+import QGroundControl.Palette           1.0
+import QGroundControl.Controls          1.0
+import QGroundControl.FlightMap         1.0
+import QGroundControl.ShapeFileHelper   1.0
+
+Item {
+    id: _root
+    property var             mapControl
+    property var             mapPolygon
+    property var             missionItems
+    property bool            interactive         : mapPolygon.interactive
+    property color           interiorColor       : "transparent"
+    property color           altColor            : "transparent"
+    property real            interiorOpacity     : 1
+    property int             borderWidth         : 0
+    property color           borderColor         : "black"
+    property string          _instructionText    : _polygonToolsText
+    property var             _StoreCoordinaates  : []
+    property real            _zorderDragHandle   : QGroundControl.zOrderMapItems + 3
+    property real            _zorderSplitHandle  : QGroundControl.zOrderMapItems + 2
+    property real            _zorderCenterHandle : QGroundControl.zOrderMapItems + 1
+    readonly property string _polygonToolsText   : qsTr("Option")
+    readonly property string _fenceText          : qsTr("Click on the map to add vertices. Click 'Done Fencing' when finished.")
+    property string          someParameter       : "defaultParameter"
+    property real            radius              : ScreenTools.defaultFontPixelHeight * 4.44     //Automatic Geofence Radius
+    property real            centerX             //Initial Drone X-Position
+    property real            centerY             //Initial Drone Y-Position
+    property bool            resetChecked
+
+    //Create Geofence Polygon Visuals
+    function addCommonVisuals() {
+        if (_objMgrCommonVisuals.empty) {
+            _objMgrCommonVisuals.createObject(polygonComponent, mapControl, true)
+        }
+    }
+
+    //Delete Geofence Polygon Visuals
+    function removeCommonVisuals() {
+        _objMgrCommonVisuals.destroyObjects()
+    }
+
+    //Create Editing tools for Polygon Visuals
+    function addEditingVisuals() {
+        if (_objMgrEditingVisuals.empty) {
+            _objMgrEditingVisuals.createObjects([ dragHandlesComponent, splitHandlesComponent, centerDragHandleComponent ], mapControl, false /* addToMap */)
```

Comment:
Don’t Use a Comment When You Can Use a Function or a Variable
Robert Martin, Clean code, page 67

![image](https://github.com/tiiuae/qgroundcontrol/assets/127403152/f0b8247e-5e3f-47ec-834b-a8c181e08b99)